src/methods.jl

```diff
@@ -1,4 +1,13 @@
-export FiniteDifferenceMethod, fdm, backward_fdm, forward_fdm, central_fdm
+export FiniteDifferenceMethod, fdm, backward_fdm, forward_fdm, central_fdm, extrapolate_fdm
+
+"""
+    add_tiny(x::Real)
+
+Add a tiny number, 10^{-40}, to `x`, preserving the type. If `x` is an `Integer`, it is
+promoted to a suitable floating point type.
+"""
+add_tiny(x::T) where T<:AbstractFloat = x + convert(T, 1e-40)
+add_tiny(x::Integer) = add_tiny(float(x))
```

Comment:
For this to work with arbitrary real types, including dimensionful types, you should probably use something like
```jl
add_tiny(x::T) where T<:Number = x + oneunit(T) * oftype(float(real(one(T))), 1e-40)
```

However, philosophically I'm very suspicious of this — not only are you assuming an absolute scale for `x`, but also you are using `1e-40` independent of the precision.

Furthermore, in most cases this code will be equivalent to `float(x)`, since the `1e-40` will be rounded off.  What are you trying to accomplish here?

Reply:
Perhaps it's better to leave support for dimensionful types for a future PR. I'm not familiar with the `Unitful` package, so I'm not sure what other things would need to change. The typing should currently be restrictive enough to throw method errors if things are not converted to `AbstractFloat`s, so there should be no infinite loops like there was before.

The purpose of `add_tiny` is to add a number that is small in the absolute sense. An example where this is used is in the step size estimation. The absolute value on the roundoff error on `x` is computed by `eps`. If `x = 0.0`, then `eps(x)` is very small, and things can NaN out. To prevent this from happening, we use `add_tiny(eps(x))` instead.

Reply:
Should we make some minimal change, if the above works we can do that,
and then later investigate unitful support more broadly.

Reply:
The above change would add for complex numbers a tiny real number to only the real part. I'm not sure what the behaviour for complex numbers should be, as tiny would then (most logically) refer to the modulus, and there are many complex numbers with modulus `1e-40`. I'm happy to go with the above change if you think that this is not important. Alternatively, we can restrict the scope of the function change the docstring of the function to only work on `AbstractFloat`s and `Integer`s; the function is only used internally anyway.

Reply:
> The purpose of add_tiny is to add a number that is small in the absolute sense.

There is no such number — there is no such thing as “small in the absolute sense”. Smallness is always relative to something else. You are implicitly assuming some absolute scale here, which cannot possibly hold for arbitrary functions. 

For example, if the function I’m differentiating is `exp10(x)` and I want the derivative at `x=-50`, then `1e-40` is not a small number in comparison to my function value. Or if I have `log10(x)` and I’m calculating the derivative at `x=1e-50`, then `1e-40` is not a small number in comparison to `x`. 

The whole point of floating-point arithmetic is that it is scale-invariant (until you reach underflow/overflow). I suspect that you are breaking that here.

Reply:
`add_tiny` has now been removed as of https://github.com/JuliaDiff/FiniteDifferences.jl/pull/125